spec/lib/guard/puma_spec.rb

```diff
@@ -61,40 +73,125 @@
         guard.start
       end
     end
+
+    describe "UI message" do
+      before do
+        allow(guard.runner).to receive(:start)
+      end
+
+      context "no :config in options" do
+        it "contains port" do
+          expect(Guard::UI).to receive(:info).with(
+            /^Guard::Puma will now start your app on port 4000 using \w+ environment.$/
+          )
+          guard.start
+        end
+      end
+
+      context ":config in options" do
+        let(:options) { { :config => 'config.rb' } }
+
+        it "doesn't contain port" do
+          expect(Guard::UI).to receive(:info).with(
+            /^Guard::Puma will now start your app using \w+ environment.$/
+          )
+          guard.start
+        end
+      end
+    end
   end
 
   describe '#reload' do
 
     before do
       expect(Guard::UI).to receive(:info).with('Restarting Puma...')
-      expect(Guard::Notifier).to receive(:notify).with(/Puma restarting/, hash_including(:image => :pending))
+      expect(Guard::UI).to receive(:info).with('Puma restarted')
       allow(guard.runner).to receive(:restart).and_return(true)
     end
 
     let(:runner_stub) { allow_any_instance_of(Guard::PumaRunner).to receive(:halt) }
 
-    it "restarts and show the message" do
-      expect(Guard::UI).to receive(:info)
-      expect(Guard::Notifier).to receive(:notify).with(/Puma restarted/, hash_including(:image => :success))
+    context "with default options" do
+      it "restarts and show the message" do
+        expect(Guard::Notifier).to receive(:notify).with(
+          /^Puma restarting on port 4000 in \w+ environment...$/,
+          hash_including(:title => "Restarting Puma...", :image => :pending)
+        )
 
-      guard.reload
+        expect(Guard::Notifier).to receive(:notify).with(
+          "Puma restarted on port 4000.",
+          hash_including(:title => "Puma restarted!", :image => :success)
+        )
+
+        guard.reload
+      end
+    end
+
+    context "with :config option" do
+      let(:options) { { :config => "config.rb" } }
+
+      it "restarts and show the message" do
+        expect(Guard::Notifier).to receive(:notify).with(
+          /^Puma restarting in \w+ environment...$/,
+          hash_including(:title => "Restarting Puma...", :image => :pending)
+        )
+
+        expect(Guard::Notifier).to receive(:notify).with(
+          "Puma restarted.",
+          hash_including(:title => "Puma restarted!", :image => :success)
+        )
+
+        guard.reload
+      end
+    end
+
+    context "with custom :notifications option" do
+      let(:options) { { :notifications => [:restarted] } }
+
+      it "restarts and show the message only about restarted" do
+        expect(Guard::Notifier).not_to receive(:notify).with(/Puma restarting/)
```

Comment:
`/restarting/` is fine, we don't need to test that `Puma` is also there. Same for the line below.

Reply:
Done.